js/ui/layout.js

```diff
@@ -361,7 +361,11 @@ const LayoutManager = new Lang.Class({
                                                            monitorIndex: monitorIndex });
 
         bgManager.connect('changed', Lang.bind(this, this._addBackgroundMenu));
-        this._addBackgroundMenu(bgManager);
+
+        // We don't want the background menu enabled on the desktop
+        // during the FBE, or in any mode without an overview, fwiw.
+        if (Main.sessionMode.currentMode.hasOverview)
```

Comment:
You can simply use `Main.sessionMode.hasOverview`, because the current mode properties are propagated to the object: https://github.com/endlessm/gnome-shell/blob/master/js/ui/sessionMode.js#L242

Reply:
https://github.com/endlessm/gnome-shell/blob/fc24fb44a2f3b3c9862f6807e5ffd175dd65c1c7/js/ui/backgroundMenu.js#L46 
This is probably a better place to put this check; that's where it was in eos-desktop, and there's another caller for the method in `backgroundMenu`.

Reply:
Both comments make sense to me (and explain why this was not a problem in the previous shell).

Updated